sssd_test_framework/utils/sssd_sockets.py

```diff
@@ -0,0 +1,278 @@
+        """
+        Check if the responder's socket unit is active.
+
+        :return: True if active, False otherwise.
+        :rtype: bool
+        """
+        result = self.host.conn.exec(["systemctl", "is-active", self.socket_unit], raise_on_error=False)
+        return result.stdout.strip() == "active"
+
+    def is_service_active(self) -> bool:
+        """
+        Check if the responder's service unit is active.
+
+        :return: True if active, False otherwise.
+        :rtype: bool
+        """
+        result = self.host.conn.exec(["systemctl", "is-active", self.service_unit], raise_on_error=False)
+        return result.stdout.strip() == "active"
+
+    def get_socket_path(self) -> Optional[str]:
+        """
+        Get the UNIX socket path for the responder.
+
+        :return: Socket path if found, None otherwise.
+        :rtype: Optional[str]
+        """
+        result = self.host.conn.exec(
+            ["systemctl", "show", self.socket_unit, "--property=Listen"], raise_on_error=False
+        )
+        if result.rc != 0:
+            raise RuntimeError(f"Failed to get socket path for {self.socket_unit}: {result.stderr}")
+
+        match = re.search(r"Listen=([^\s]+)", result.stdout)
+        return match.group(1) if match else None
+
+
+class SSSDSocketUtils(MultihostUtility[MultihostHost]):
+    def __init__(self, host: MultihostHost) -> None:
+        super().__init__(host)
+
+    @property
+    def nss(self) -> SSSDSocketResponder:
+        return SSSDSocketResponder(self.host, ResponderType.NSS)
+
+    @property
+    def pam(self) -> SSSDSocketResponder:
+        return SSSDSocketResponder(self.host, ResponderType.PAM)
+
+    @property
+    def sudo(self) -> SSSDSocketResponder:
+        return SSSDSocketResponder(self.host, ResponderType.SUDO)
+
+    @property
+    def ssh(self) -> SSSDSocketResponder:
+        return SSSDSocketResponder(self.host, ResponderType.SSH)
+
+    @property
+    def pac(self) -> SSSDSocketResponder:
+        return SSSDSocketResponder(self.host, ResponderType.PAC)
+
+    @property
+    def autofs(self) -> SSSDSocketResponder:
+        return SSSDSocketResponder(self.host, ResponderType.AUTOFS)
+
+    def remove_services_from_sssd_conf(self, socket_responders: Optional[list[str]] = None) -> None:
+        """
+        Remove only the socket-activated responders from the 'services' line in sssd.conf.
+        Remove the entire line if it becomes empty after removal.
+
+        :param socket_responders: List of responders to remove (default: all known).
+        """
+        socket_responders = socket_responders or [
+            ResponderType.NSS,
+            ResponderType.PAM,
+            ResponderType.SUDO,
+            ResponderType.SSH,
+            ResponderType.PAC,
+            ResponderType.AUTOFS,
+        ]
+
+        # Extract current services line
+        result = self.host.conn.exec(["grep", "^services[[:space:]]*=", "/etc/sssd/sssd.conf"], raise_on_error=False)
```

Comment:
I think there is a dedicated class to manage sssd config.
Note that there are might be snippets, etc.
At the very least a path to config file shouldn't be hardcoded.